### PR TITLE
[#86] Add extra image validation

### PR DESCRIPTION
**Pull request by krissy**
**_Addresses backend component of #86**_
# What this does
- [x] Add extra image validation to restrict upload images to the following formats: .jpg, .gif, .png, .tiff, .png
- [x] Add clarification of acceptable formats in validation message
- [x] Add some tests for image validation
- [x] Converts all image uploads to JPEGs going forward to simplify things and avoid TIFF issues
## Screenshots

**Validation on upload attempt of unsupported format:**

screen shot 2016-08-21 at 14 56 55

**Green placeholder preview when tiff uploaded (not yet converted to jpg):**

screen shot 2016-08-21 at 16 56 38

### CODE REVIEW

**Comment by steverabino** [comment]
:+1: :+1: :+1:

**Comment by CathMollie** [comment]
@krissy Just noting tiff upload wasn't very successful in usability test - took a long time and displayed as broken link on 'manage drawings' page.

**Comment by steverabino** [comment]
Good save @CathMollie, Kris is going to do some conversion magic!

**Comment by krissy** [comment]
@CathMollie good to know! Tanya just tested and you're right. Steve found this could be an existing issue with the file management tool we're using so working on a fix as part of this branch.

**Comment by krissy** [comment]
Tiff issue fixed in 11e5bfc - please review. Now all images will be converted to jpgs going forward

File: app/models/drawing.rb [hunk 1]
```diff
@@ -16,8 +16,9 @@ class Drawing < ActiveRecord::Base
   validates :mood_rating, numericality: { only_integer: true, greater_than_or_equal_to: 1, less_than_or_equal_to: 10 }, allow_nil: true
 
   has_attached_file :image, styles: {
-    medium: "640x",
-    thumb: "100x100#"
+    medium: ["640x", :jpg],
+    thumb: ["100x100#", :jpg],
+    original: ["100%", :jpg]
```
Comment (steverabino) [inline app/models/drawing.rb:21]: Good idea, can link to this from the show.html.haml page!

**Comment by stevesmith2609** [comment]
:+1:

**Comment by steverabino** [comment]
FYI @krissy for tiff, the preview isn't working; I guess the image hasn't been converted by this point.

Is there a way to do the conversion before showing the preview; or shall we not worry about this for now?

image

**Comment by krissy** [comment]
TIFF preview addressed in 39aedc3 and screenshot added to PR description. Browsers don't support TIFFs without a hacky js library, and images aren't saved (and thus converted to jpegs) until the drawing is saved. So for TIFFs we'll show an alternate green placeholder to indicate change.

On a side note, we need a separate ticket to add some client-side validation for supported image formats. E.g. if PDF upload attempted, we show a red cross placeholder and appropriate validation message reminding user about supported formats.

**Comment by steverabino** [comment]
:+1: Nice work - sorry I found the bug :(

**Pull request merged**